### ISSUE TITLE
Different architectures for MNIST and celebA

### ISSUE THREAD

**Issue by affromero** [carpedm20/DCGAN-tensorflow#217]
Hello,
- Why are there two different architectures (G and D) for the two datasets? Is there any special reason or you just came up with them?

- Why do you need to concatenate labels (MNIST) every layer in the Generator? [https://github.com/carpedm20/DCGAN-tensorflow/blob/master/model.py#L437-L449](https://github.com/carpedm20/DCGAN-tensorflow/blob/master/model.py#L437-L449)

- Why `tanh` for celebA and `sigmoid` for MNIST?

Thanks!

**Comment by wenzishou** [comment]
can anyone explain the reason?

**Comment by affromero** [comment]
@wenzishou After being working with GANs for a while, I could answer my own old question:

- DCGAN generates 64x64 images for CelebA, and 28x28 images for MNIST. Well, the same network can't do it. 

- This is just for stabilization in the training framework. I *guess* the main reason is that the labels are not ignored by G in this scheme.  

- This is related to the input of the network regarding the dataset. CelebA is normalized [-1,1] (tanh) and MNIST [0,1] (sigmoid). 

Hope it helps you.

**Comment by affromero** [comment]
- DCGAN

**Comment by wenzishou** [comment]
@affromero The paper of DCGAN doesn't say the specific process of MNIST, do you heve any references of this framework
